src/Microsoft.OData.Core/SelectedPropertiesNode.cs

```diff
@@ -582,19 +583,21 @@ private bool IsOperationSelectedAtThisLevel(IEdmOperation operation, bool mustBe
 
         /// <summary>Create SelectedPropertiesNode from SelectExpandClause.</summary>
         /// <param name="selectExpandClause">The SelectExpandClause representing $select and $expand clauses.</param>
+        /// <param name="version">OData version</param>
         /// <returns>SelectedPropertiesNode generated using <paramref name="selectExpandClause"/></returns>
-        private static SelectedPropertiesNode CreateFromSelectExpandClause(SelectExpandClause selectExpandClause)
+        private static SelectedPropertiesNode CreateFromSelectExpandClause(SelectExpandClause selectExpandClause, ODataVersion version)
         {
             SelectedPropertiesNode node;
-            selectExpandClause.Traverse(ProcessSubExpand, CombineSelectAndExpandResult, out node);
+            selectExpandClause.Traverse(ProcessSubExpand, CombineSelectAndExpandResult, version, out node);
             return node;
         }
 
         /// <summary>Process sub expand node, set name for the node.</summary>
         /// <param name="nodeName">Node name for the subexpandnode.</param>
         /// <param name="subExpandNode">Generated sub expand node.</param>
+        /// <param name="version">OData version.</param>
         /// <returns>The sub expanded node passed in.</returns>
-        private static SelectedPropertiesNode ProcessSubExpand(string nodeName, SelectedPropertiesNode subExpandNode)
+        private static SelectedPropertiesNode ProcessSubExpand(string nodeName, SelectedPropertiesNode subExpandNode, ODataVersion version)

```

Comment:
>, ODataVersion version [](start = 116, length = 22)

This newly added "version" argument is not referenced/used in this function?

Reply:
No, but we want to pass this function as the first parameter to SelectExpandClause.Traverse, which takes a Func<string, SelectedPropertiesNode, ODataVersion); it's not used in this case, but it is used in other cases so it needs to be part of the signature.

---
In reply to: [175302563](https://github.com/OData/odata.net/pull/1109#discussion_r175302563) [](ancestors = 175302563)